src/ServiceInsight/Search/SearchBarViewModel.cs

```diff
@@ -52,22 +51,22 @@ protected override void OnActivate()
 
         public void GoToFirstPage()
         {
-            Parent.RefreshMessages(SelectedEndpoint, 1, SearchQuery);
+            Parent.RefreshMessages(FirstLink);
         }
 
         public void GoToPreviousPage()
         {
-            Parent.RefreshMessages(SelectedEndpoint, CurrentPage - 1, SearchQuery);
+            Parent.RefreshMessages(PrevLink);
         }
 
         public void GoToNextPage()
         {
-            Parent.RefreshMessages(SelectedEndpoint, CurrentPage + 1, SearchQuery);
+            Parent.RefreshMessages(NextLink);
         }
 
         public void GoToLastPage()
         {
-            Parent.RefreshMessages(SelectedEndpoint, PageCount, SearchQuery);
+            Parent.RefreshMessages(LastLink);
```

Comment:
I must say I preferred the previous name "NavigateTo" but that is just my taste

Reply:
That's also refactoring we could consider. It doesn't have to be part of this PR.